### ISSUE TITLE
Templates: Hide description field when switching to List layout

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#57596]
Since template descriptions can sometimes be quite long, they are not handled beautifully in data views list layout:

<img width="752" alt="Screenshot 2024-04-16 at 11 50 42" src="https://github.com/WordPress/gutenberg/assets/846565/e84db1b7-1edd-4d34-8db5-1f093c25680c">

When switching from the default layout (Grid) to List, it might be good to auto-hide the description field.

<details><summary>Original issue</summary>In the template management data view – list layout – the description isn't particularly useful for two main reasons:

1. Most descriptions are too long to be fully visible
2. Some templates do not have descriptions at all, resulting in an unhelpful `–` which takes up a lot of vertical space

<img width="1079" alt="Screenshot 2024-01-05 at 14 22 53" src="https://github.com/WordPress/gutenberg/assets/846565/c5b99b4e-b726-4909-a4c0-2617b7022d9b">

Removing the description would reduce a lot of noise and save space:

<img width="997" alt="Screenshot 2024-01-05 at 14 24 33" src="https://github.com/WordPress/gutenberg/assets/846565/0a529508-a776-4b12-a80b-2f3688428821"></details>

**Comment by carolinan** [comment]
I don't think removing them is the best solution. We know that users are still struggling with concepts of index, home, and front page. The descriptions definitely have value for these users.

I am not able to see where this vertical layout is used? How do I access it? Did you mean to only hide it on narrow screens?

**Comment by jameskoster** [comment]
> I am not able to see where this vertical layout is used?

Enable the "New admin views" experiment in Gutenberg then navigate to Appearance > Editor > Templates > Manage all templates. From there change the layout to "List" in the view options. 

Since this view is optimised for quickly browsing records in a more visual manner, I don't know that the description is required _in the list_. It can be accessed easily once it's possible to reach the details panel for each record (see https://github.com/WordPress/gutenberg/pull/57523).

**Comment by jameskoster** [comment]
I think this is much less of an issue since Grid was made the default layout for templates, but it would be good to auto-hide the description when switching from grid to list layout.

**Comment by jasmussen** [comment]
If you can invoke the normalized details panel / inspector in this view, then you'd be able to see the description there.

**Comment by jameskoster** [comment]
Yup. You could also manually expose the description in view options if you really need to see it in this layout.

**Comment by Souptik2001** [comment]
Hi @jameskoster @jasmussen ,

I have raised a PR for this. Please review at your convenience.

Just adding one point over here -

As you can see in the below video -
- Suppose the description field is not hidden.
- And now I switch to the "List" layout and the description field becomes auto-hidden.
- Now when I switch to any other layout the description field still remains hidden.
- This is ok right?

https://github.com/WordPress/gutenberg/assets/39427067/04717273-3d55-4ac5-a2af-1d6e507e3179

**Comment by jameskoster** [comment]
> Suppose the description field is not hidden.
> And now I switch to the "List" layout and the description field becomes auto-hidden.
> Now when I switch to any other layout the description field still remains hidden.
> This is ok right?

This is a good question because it may indicate we need a more fundamental change to get this right.

I'm keen to hear feedback from others but my feeling is that in this example switching from List layout to any other layout should reveal the description again. 

What makes sense to me:

* Data views consumers can specify layout-specific configs for field visibility. E.g. the Templates page might define the `description` field as visible by default on Table layout, but hidden by default on List and Grid layouts.
* Any user customisation of field visibility takes priority over the default config.

So as a working example, on the templates page: Switching from Table to List layout would hide the description. Switching back to Table layout would reveal it again. However, if a user toggled the description field _off_ in Table layout, then switched to List layout, then _back_ to Table layout, the description would remain hidden. This seems reasonable to me because the user made an active decision around field visibility, and that should persist for the current session.

What do you think?

**Comment by ellatrix** [comment]
This is an enhancement, so punting to WP 6.7.

**Comment by youknowriad** [comment]
Yes, it seems to me, the fundamental issue is that we want by default different fields for different layouts and it might not make sense to retain the fields when you switch layouts.

Folks want to retain things like "filters" and "search" but maybe it's not the case for the rest (pagination: arguable, fields)
